File: spec/functional_test/fixtures/python/flask_foreign/service.py
# A FastAPI handler file in the same repo. It uses `@app.get`/`@app.post`
# decorators and never mentions flask. The Flask analyzer must NOT claim
# it (mislabeling its routes python_flask); the FastAPI analyzer owns it.
from fastapi import FastAPI

app = FastAPI()


@app.get("/fastapi-items")
def list_items():
    return []


@app.post("/fastapi-items")
def create_item(name: str):
    return {"name": name}

File: spec/functional_test/fixtures/javascript/express_hono_foreign/service.ts
// A Hono handler file in the same repo. It uses the same `.get()/.post()`
// chaining shape Express's shared route extractor recognizes, but never
// mentions express. The Express analyzer must NOT claim it (mislabeling
// its routes js_express); the Hono analyzer owns it (#2368).
import { Hono } from 'hono'

const app = new Hono()

app.get('/hono-items', (c) => c.json([]))
app.post('/hono-items', (c) => c.json({}))

export default app
